src/olympia/lib/crypto/signing.py

```diff
@@ -53,19 +55,79 @@ def get_id(addon):
     return hashlib.sha256(guid).hexdigest()
 
 
-def call_signing(file_obj):
-    """Get the jar signature and send it to the signing server to be signed."""
-    # Extract jar signature.
+def autograph_sign_file(file_obj):
+    """Sign `file_obj` via autographs /sign/file endpoint.
+
+    :returns: The certificates serial number.
+    """
+    conf = settings.AUTOGRAPH_CONFIG
+
+    with storage.open(file_obj.current_file_path) as fobj:
```

Comment:
You should open in `rb` mode I think since you want to encode in base64, which takes bytes.

Reply:
Then again this probably doesn't matter in Py2 and you've got more potential for breakage through other calls below that deal with bytes in various ways... So maybe don't bother for now, it will have to be fixed in https://github.com/mozilla/addons-server/issues/10423

Reply:
Perfect, yeah, happy to tackle that as part of #10423 then